src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -221,24 +221,36 @@ private static object PtrToStructureHelper(IntPtr ptr, Type structureType)
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern bool IsPinnable(object? obj);
 
+        internal static bool IsComSupported { get; } = InitializeIsComSupported();
+
+        private static bool InitializeIsComSupported() =>
+            AppContext.TryGetSwitch("System.Runtime.InteropServices.Marshal.IsComSupported", out bool isSupported) ? isSupported : true;
+
 #if FEATURE_COMINTEROP
         /// <summary>
         /// Returns the HInstance for this module.  Returns -1 if the module doesn't have
         /// an HInstance.  In Memory (Dynamic) Modules won't have an HInstance.
         /// </summary>
         public static IntPtr GetHINSTANCE(Module m)
```

Comment:
This API has nothing to do with COM. It is under `FEATURE_COMINTEROP` by accident.